src/lib/operators/join_hash.cpp

```diff
@@ -25,9 +25,10 @@ namespace opossum {
 JoinHash::JoinHash(const std::shared_ptr<const AbstractOperator>& left,
                    const std::shared_ptr<const AbstractOperator>& right, const JoinMode mode,
                    const ColumnIDPair& column_ids, const PredicateCondition predicate_condition,
-                   const std::optional<size_t>& radix_bits)
+                   const std::optional<size_t>& radix_bits, std::vector<JoinPredicate> additional_join_predicates)
```

Comment:
The sort merge join is still outdated. I prefer passing it by value because now it is up to the caller to decide if he still requires the vector or not.

Reply:
Why not a simple reference?

Reply:
To my understanding the reasoning is as follows:
If we accepted a simple reference we would call the copy constructor when initializing our member variable. By accepting the vector as a value it is up to the caller to decide if he still requires the vector. If not, he can just move it and we save one copy operation. Otherwise we would always have that copy operation.